src/server/parser.rs

```diff
@@ -931,4 +816,185 @@ mod test {
         });
     }
 
+    #[test]
+    fn test_newline_delimited() {
+        let mut io = MemIo::new();
+        let mut lp = MockLoop::new(Default::default());
+        io.push_bytes("GET / HTTP/1.1\n\
+            Content-Length: 0\n\
+            Connection: close\n\n"
+                          .as_bytes());
+        let m = Stream::<Parser<Proto, MemIo>>::accepted(io.clone(), &mut lp.scope(1))
+                    .expect_machine();
+        m.ready(EventSet::readable(), &mut lp.scope(1))
+         .expect_machine();
+        assert_eq!(*lp.ctx(),
+                   Context {
+                       progressive: false,
+                       headers_received: 1,
+                       body: String::from(""),
+                       chunks_received: 0,
+                       requests_received: 1,
+                   });
+    }
+
+    // FIXME
+    #[test]
+    #[should_panic]
```

Comment:
It parses just one because `Proto` doesn't write anything to the response. And parser needs `Response` to be fully done before it starts to parse next request.